projects/mongoose/build.sh

```diff
@@ -13,5 +13,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
 cd $SRC/mongoose
 $CXX $CXXFLAGS $LIB_FUZZING_ENGINE -DMG_ENABLE_LINES -DMG_ENABLE_LOG=0 mongoose.c -I. test/fuzz.c -o $OUT/fuzz
+
+# Fuzzer using honggfuzz netdriver.
+if [[ "$FUZZING_ENGINE" == "honggfuzz" ]]
+then
+  export HONGGFUZZ_HOME=$SRC/honggfuzz
+  $HONGGFUZZ_HOME/hfuzz_cc/hfuzz-clang $CFLAGS -DMG_ENABLE_LINES=1 \
```

Comment:
Can we provide HFUZZ_CC/CXX, LIB_HF_NETDRIVER env vars from the base builder to be used here instead? 

Also, is it possible at all to avoid `hfuzz-clang` and use clang directly here?

Reply:
This is just coming from how we can simplify integration into upstream build systems. The less custom stuff we need to pass on top of our existing CC/CXX, CFLAGS etc, the better.

Reply:
We can avoid hfuzz-clang but we need some macros in there if we do that. I added the logic in the `compile_honggfuzz` so now one has to use `$LIB_HF_NETDRIVER "$HF_ND_CC" and `$LIB_HF_NETDRIVER "$HF_ND_CXX"`
for C and C++ honggfuzz netdriver targets. 

WDYT of the current state?